packages/core/components/MenuBar/index.tsx

```diff
@@ -77,22 +78,25 @@ export const MenuBar = ({
           </IconButton>
         </div>
         <>
-          {renderHeaderActions &&
+          {renderHeaderActions ? (
             renderHeaderActions({
               state: appState,
               dispatch,
-            })}
+              onPublish: () => onPublish && onPublish(data),
+            })
+          ) : (
+            <div>
+              <Button
+                onClick={() => {
+                  onPublish && onPublish(data);
+                }}
+                icon={<Globe size="14px" />}
+              >
+                Publish
+              </Button>
+            </div>
```

Comment:
This should be passed as `children` to `renderHeaderActions` and `overrides.headerActions`, so the user can still choose to render the default button

Reply:
From this I understand you want me to rewrite the component in this way.

```
{renderHeaderActions && (
            renderHeaderActions({
              state: appState,
              dispatch,
              onPublish: () => onPublish && onPublish(data),
            })
          ) 
```

And Completely remove this button code as we can directly pass publish button as children to `renderHeaderActions`

Reply:
Sorry for the delay! Xmas etc :)

Not quite, I want something like this 

```tsx
const PublishButton = ( { onPublish } ) => (
  <div>
    <Button
      onClick={() => {
        onPublish && onPublish(data);
      }}
      icon={<Globe size="14px" />}
    >
      Publish
    </Button>
  </div>
);
```

```tsx
          {renderHeaderActions ? (
             renderHeaderActions({
               state: appState,
               dispatch,
               children: <PublishButton onPublish={onPublish} />
             </div>
             })
           ) : (
             <PublishButton onPublish={onPublish} />
           )
```

and updating the [`<CustomHeaderActions>` call](https://github.com/measuredco/puck/blob/f2f0c9d00f08b28da765abef8887407d47deb96b/packages/core/components/Puck/index.tsx#L405) to take children

```tsx
                        <CustomHeaderActions>
                          <PublishButton onPublish={onPublish} />
                        </CustomHeaderActions>
```

I don't think there's any point in passing in the `onPublish` function either.